test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -88,8 +91,14 @@
      */
     private static final String REST_TESTS_VALIDATE_SPEC = "tests.rest.validate_spec";
 
-    private static final String TESTS_PATH = "/rest-api-spec/test";
-    private static final String SPEC_PATH = "/rest-api-spec/api";
+    /**
+     * Property that allows to set the root for the rest API spec. This value plus SPEC_PATH is the location of the REST API spec can be
+     * found.
+     */
+    public static final String REST_SPEC_ROOT = "tests.rest.spec_root";
```

Comment:
It's not clear what this property is doing. We seem to be adding a way to specifify _where_ specifically in the classpath to look for this stuff, but everywhere I can tell we simply set this location to the root of the resources output directory. This seems superfluous. If we are going to continue to load these off the classpath, we should just assume they live in some particular locaiton (which we do, at `/rest-api-spec`). We are now just having to set an additional system property for all our tests.

Reply:
It flexes between the [rootBuildDir](https://github.com/elastic/elasticsearch/pull/51794/files#diff-b0cdf22e21c1c8991409f3d6056b1c01R75) and the test.output.resoureDir, depending on the test. I think with the refactoring suggested above, it may be able to eliminate the variance and just use a well known path.

Reply:
I'm a bit confused, are we adding `rootBuildDir` to the classpath? As far as I can tell we are still loading these things from the classpath, not from the filesystem, or did I miss something?

Reply:
to close the loop here... 
no, the rootBuildDir was not added to the classpath, the api specs (not the tests) in this pr are read from the [file system](https://github.com/elastic/elasticsearch/pull/51794/files#diff-7a30c014b9fbbf6375d1ed5c2d67b2b3R84) which is why the [security policy change](https://github.com/elastic/elasticsearch/pull/51794/files#diff-5f27a4ae4da2cbab92dd569334cd2fecR113) was required. The refactoring you suggested removes the need for this. Update coming soon.